src/keyring.rs

```diff
@@ -1,17 +1,48 @@
+//! Keyring to perform rpgp operations with.
+
 use anyhow::Result;
 
-use crate::constants::KeyType;
 use crate::context::Context;
-use crate::key::Key;
+use crate::key::{self, DcKey};
 
-#[derive(Default, Clone, Debug)]
-pub struct Keyring {
-    keys: Vec<Key>,
+/// An in-memory keyring.
+///
+/// The keyring can contain both owned and borrowed keys to avoid unnecessary
+/// copies as most keyring are constructed just for the rpgp operation and
```

Comment:
it only contains owned values now, could you update the description, I forgot about that